docs/flutterweathertutorial.md

```diff
@@ -43,476 +67,2514 @@ We'll be focusing on two endpoints:
-
-Awesome! We now have an app that looks way nicer (in my opinion :P) and have tackled problem #2.
-
-All that's left is to handle unit conversion between celsius and fahrenheit. To do that we'll create a `Settings` widget and a `SettingsBloc`.
-
-## Unit Conversion
-
-We'll start off by creating our `SettingsBloc` which will convert `SettingsEvents` into `SettingsStates`.
-
-Our `SettingsEvents` will consist of a single event: `TemperatureUnitsToggled`.
-
-[settings_event.dart](_snippets/flutter_weather_tutorial/settings_event.dart.md ':include')
-
-Our `SettingsState` will simply consist of the current `TemperatureUnits`.
-
-[settings_state.dart](_snippets/flutter_weather_tutorial/settings_state.dart.md ':include')
-
-Lastly, we need to create our `SettingsBloc`:
-
-[settings_bloc.dart](_snippets/flutter_weather_tutorial/settings_bloc.dart.md ':include')
-
-All we're doing is using `fahrenheit` if `TemperatureUnitsToggled` is added and the current units are `celsius` and vice versa.
-
-Now we need to provide our `SettingsBloc` to our `App` widget in `main.dart`.
-
-[main.dart](_snippets/flutter_weather_tutorial/main4.dart.md ':include')
-
-Again, we're making `SettingsBloc` globally accessible using `BlocProvider` and we are also closing it in the `close` callback. This time, however, since we are exposing more than one Bloc using `BlocProvider` at the same level we can eliminate some nesting by using the `MultiBlocProvider` widget.
-
-Now we need to create our `Settings` widget from which users can toggle the units.
-
-[settings.dart](_snippets/flutter_weather_tutorial/settings.dart.md ':include')
-
-We're using `BlocProvider` to access the `SettingsBloc` via the `BuildContext` and then using `BlocBuilder` to rebuild our UI based on `SettingsState` changed.
-
-Our UI consists of a `ListView` with a single `ListTile` which contains a `Switch` that users can toggle to select celsius vs. fahrenheit.
-
-?> **Note:** In the switch's `onChanged` method we add a `TemperatureUnitsToggled` event to notify the `SettingsBloc` that the temperature units have changed.
-
-Next, we need to allow users to get to the `Settings` widget from our `Weather` widget.
+Open [https://www.metaweather.com/api/location/search/?query=london](https://www.metaweather.com/api/location/search/?query=london) in your browser to see the response for the city of London. We will use the `woeid` (where-on-earth-id) in the return dictionary to hit the location endpoint.
+
+The `woeid` for London is 44418. Navigate to [https://www.metaweather.com/api/location/44418](https://www.metaweather.com/api/location/44418) in your browser and you'll see the response for weather in London. This contains all the data we will need for our app.
+
+### Creating Our MetaWeather API Client Subpackage
+
+> Since our API client interacting MetaWeather should be independent of the rest of our app, we can create it as a subpackage. We can then import our package in our `pubspec.yaml`
```

Comment:
Can we provide a reason for this? Is it readability or a best practice? Maybe something like: 

We want to keep our API client independent of the rest of the app [for x reason(s)]... We can create it as a subpackage and then import it in our `pubspec.yaml`